packages/backend-storage/src/factory.ts

```diff
@@ -56,19 +58,29 @@ class AmplifyStorageFactory
     }
     return constructContainer.getOrCompute(this.generator) as AmplifyStorage;
   };
+
+  private validateName = (name: string): void => {
+    const nameIsAlphanumeric = /^[a-zA-Z0-9]+$/.test(name);
+    if (!nameIsAlphanumeric) {
+      throw new AmplifyUserError('InvalidResourceNameError', {
+        message: `defineStorage name can only contain alphanumeric characters, found ${name}`,
+        resolution:
+          'Change the name parameter of defineStorage to only use alphanumeric characters',
+      });
+    }
+  };
 }
 
 class AmplifyStorageGenerator implements ConstructContainerEntryGenerator {
   readonly resourceGroupName = 'storage';
-  private readonly defaultName = 'amplifyStorage';
 
   constructor(
     private readonly props: AmplifyStorageProps,
     private readonly outputStorageStrategy: BackendOutputStorageStrategy<BackendOutputEntry>
   ) {}
 
   generateContainerEntry = (scope: Construct) => {
-    return new AmplifyStorage(scope, this.defaultName, {
+    return new AmplifyStorage(scope, `${this.props.name}`, {
```

Comment:
Do we no longer want amplifyStorage in the name?

Reply:
what if name is undefined

Reply:
Name is required in props. I don't think we need any sort of `amplifyStorage` prefix because presumably the customer will recognize where the name came from since they defined it